src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.cpp

```diff
@@ -0,0 +1,257 @@
+
+        // If solid stress solver has been called, call updateMeanSolidStress to project stress state onto faults
+        if ( subRegion.hasWrapper( SolidMechanicsLagrangianFEM::viewKeyStruct::solidMaterialNamesString() ) )
+        {
+          initializeFaultOrientation();
+
+          updateMeanSolidStress( subRegion );
+          forAll< parallelDevicePolicy<> >(  sig.size(), [=] GEOS_HOST_DEVICE ( localIndex const k )
+          {
+            // Set initial stress conditions on faults
+            sig_i[k] = sig[k];
+            tau_i[k] = tau[k];
+          } );
+
+        }
+        // If solid stress solver has not been called, initialize fault stresses to user-defined values
+        else
+        {
+          arrayView1d< real64 > const tempSigIni = subRegion.getField< inducedSeismicity::initialMeanNormalStress >();
+          tempSigIni.setValues< parallelHostPolicy >( m_initialFaultNormalStress );
+          arrayView1d< real64 > const tempSig = subRegion.getField< inducedSeismicity::meanNormalStress >();
+          tempSig.setValues< parallelHostPolicy >( m_initialFaultNormalStress );
+          arrayView1d< real64 > const tempSig_n = subRegion.getField< inducedSeismicity::meanNormalStress_n >();
+          tempSig_n.setValues< parallelHostPolicy >( m_initialFaultNormalStress );
+
+          arrayView1d< real64 > const tempTauIni = subRegion.getField< inducedSeismicity::initialMeanShearStress >();
+          tempTauIni.setValues< parallelHostPolicy >( m_initialFaultShearStress );
+          arrayView1d< real64 > const tempTau = subRegion.getField< inducedSeismicity::meanShearStress >();
+          tempTau.setValues< parallelHostPolicy >( m_initialFaultShearStress );
+          arrayView1d< real64 > const tempTau_n = subRegion.getField< inducedSeismicity::meanShearStress_n >();
+          tempTau_n.setValues< parallelHostPolicy >( m_initialFaultShearStress );
+        }
+      } );
+    } );
+  }
+}
+
+void SeismicityRateBase::postProcessInput()
+{
+  // Initialize member stress solver as specified in XML input
+  m_stressSolver = &this->getParent().getGroup< SolverBase >( m_stressSolverName );
+
+  SolverBase::postProcessInput();
+}
+
+void SeismicityRateBase::initializeFaultOrientation()
+{
+  if ( m_faultNormal.size()!=3 || m_faultShear.size()!=3 )
+  {
+    // THROW ERROR, FAULT ORIENTATION MUST BE DEFINED FOR SOLID STRESS SOLVER
+    GEOS_ERROR("Proper fault orientation (faultNormal and faultShear) must be defined for seismicity rate solver if a solid stress solver is being used.");
+  }
+  else
+  {
+    if( !checkFaultOrthogonality() ){
+      GEOS_ERROR("Fault normal and fault shear must be orthogonal");
+    }
+
+    normalizeFaultVecs(); 
+
+    // Voigt notation of dyadic product of fault normal vectors 
+    // when multiplied in double dot product with symmetric stress tensor
+    m_faultNormalVoigt[0] = m_faultNormal[0]*m_faultNormal[0];
```

Comment:
I might call these `m_faultNormalProjectionTensor` and `m_faultShearProjectionTensor` to emphasize there are tensors and not vectors.